packages/synapse-react-client/test/containers/LoginMethodButton.test.tsx

```diff
@@ -0,0 +1,123 @@
+import {
+  CURRENT_SOURCEAPP_NAME_LOCALSTORAGE_KEY,
+  LOGIN_OAUTH2_GOOGLE,
+  LOGIN_USERNAME_PASSWORD,
+} from '../../src/utils/SynapseConstants'
+import { SynapseContextType } from '../../src/utils/context/SynapseContext'
+import { createWrapper } from '../testutils/TestingLibraryUtils'
+
+const onClick = jest.fn()
+
+const defaultProps: LoginMethodButtonProps = {
+  loginMethod: LOGIN_OAUTH2_GOOGLE,
+  iconName: 'google24',
+  onClick: onClick,
+}
+const defaultButtonText = 'Sign in with Google'
+const defaultCurrentAppSourceName = 'AD Knowledge Portal'
+const defaultLastLoginInfo = {
+  lastLoginMethod: LOGIN_USERNAME_PASSWORD,
+  lastLoginDate: 'May 10, 2023',
+  lastLoginSourceAppName: 'ARK Portal',
+  lastLoginSourceAppURL: 'https://arkportal.synapse.org',
+}
+
+function renderComponent(
+  props: LoginMethodButtonProps,
+  wrapperProps?: SynapseContextType,
+) {
+  return render(<LoginMethodButton {...props} />, {
+    wrapper: createWrapper(wrapperProps),
+  })
+}
+
+function setUp(
+  lastLoginInfo = defaultLastLoginInfo,
+  currentSourceApp = defaultCurrentAppSourceName,
+  props: LoginMethodButtonProps = defaultProps,
+  wrapperProps?: SynapseContextType,
+) {
+  setLastLoginInfo(
+    lastLoginInfo.lastLoginMethod as LoginMethod,
+    lastLoginInfo.lastLoginDate,
+    lastLoginInfo.lastLoginSourceAppName,
+    lastLoginInfo.lastLoginSourceAppURL,
+  )
+  localStorage.setItem(
+    CURRENT_SOURCEAPP_NAME_LOCALSTORAGE_KEY,
+    currentSourceApp,
+  )
+
+  const user = userEvent.setup()
+  const component = renderComponent(props, wrapperProps)
+  const button = screen.getByRole('button')
+  const subText = within(button).queryByText(
+    /you may be able to use an existing account/i,
+  )
+  return { component, user, button, subText }
+}
+
+describe('LoginMethodButton tests', () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
+
+  afterEach(() => {
+    clearLastLoginInfo()
+  })
+
+  it('displays friendly button text and specified logo', () => {
+    const { button } = setUp()
+    expect(button).toHaveTextContent(defaultButtonText)
+    expect(document.getElementById('icon-google24')).toBeInTheDocument()
+  })
+
+  it('calls the callback function on click', async () => {
+    const { button, user } = setUp()
+    user.click(button)
+
+    await waitFor(() => {
+      expect(onClick).toHaveBeenCalledTimes(1)
+    })
+  })
+
+  it('does not display help text when previous and current login method are different', () => {
+    const { subText } = setUp()
+    expect(subText).not.toBeInTheDocument()
+  })
+
+  it('does not display help text when apps are the same, even if previous and current login methods are the same', () => {
+    const { button, subText } = setUp(
+      { ...defaultLastLoginInfo, lastLoginMethod: LOGIN_OAUTH2_GOOGLE },
+      'ARK Portal',
+    )
+    expect(subText).not.toBeInTheDocument()
+    expect(button.querySelector('div')).toBeFalsy()
```

Comment:
Ensure that when the subtext isn't rendered, that the divs used to contain the last login info sentence also aren't rendered

Reply:
Consider using `data-testid`/`queryByTestId` for this very specific check (who knows, maybe our button icons will be have divs someday, or something), but I'm ok with how you have it.

Reply:
Good call, updated in this [commit](https://github.com/Sage-Bionetworks/synapse-web-monorepo/pull/369/commits/8ad84417cd4f641cdb3a5bd65bee2a3939dee364)